en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md

```diff
@@ -252,130 +272,142 @@ You can also try to enable it manually:
-
-eth0    Link encap:Ethernet HWaddr xx:xx:xx:xx:xx:xx at UP
-        inet addr:169.254.21.184 DRaddr:169.254.21.1 Mask:255.255.255.0
+PING 10.41.10.1 56 bytes of data
+56 bytes from 10.41.10.1: icmp_seq=0 time=0.0 ms
+56 bytes from 10.41.10.1: icmp_seq=1 time=0.0 ms
+56 bytes from 10.41.10.1: icmp_seq=2 time=0.0 ms
+56 bytes from 10.41.10.1: icmp_seq=3 time=0.0 ms
+56 bytes from 10.41.10.1: icmp_seq=4 time=0.0 ms
+56 bytes from 10.41.10.1: icmp_seq=5 time=0.0 ms
+56 bytes from 10.41.10.1: icmp_seq=6 time=0.0 ms
+56 bytes from 10.41.10.1: icmp_seq=7 time=0.0 ms
+56 bytes from 10.41.10.1: icmp_seq=8 time=0.0 ms
+56 bytes from 10.41.10.1: icmp_seq=9 time=0.0 ms
+10 packets transmitted, 10 received, 0% packet loss, time 10010 ms
+rtt min/avg/max/mdev = 0.000/0.000/0.000/0.000 ms
 ```
 
-Now the devices should be able to ping each other.
-
-Note that this configuration is ephemeral and will be lost after a reboot, so we’ll need to find a way to configure it statically.
+#### MAVLink/MAVSDK Test
 
-#### Ping Test
+For this, we need to set the MAVLink instance to send traffic to the CM4's IP address:
 
-First from the CM4:
+For an initial test we can do:
 
 ```sh
-ping 169.254.21.184
-
-PING 169.254.21.184 (169.254.21.184) 56(84) bytes of data.
-64 bytes from 169.254.21.184: icmp_seq=1 ttl=64 time=0.188 ms
-64 bytes from 169.254.21.184: icmp_seq=2 ttl=64 time=0.131 ms
-64 bytes from 169.254.21.184: icmp_seq=3 ttl=64 time=0.190 ms
-64 bytes from 169.254.21.184: icmp_seq=4 ttl=64 time=0.112 ms
-^C
---- 169.254.21.184 ping statistics ---
-4 packets transmitted, 4 received, 0% packet loss, time 3077ms
-rtt min/avg/max/mdev = 0.112/0.155/0.190/0.034 ms
+mavlink start -o 14540 -t 10.41.10.1
 ```
 
-Then from the flight controller in NuttShell:
+This will send MAVLink traffic on UDP to port 14540 (the MAVSDK/MAVROS port) to that IP which means MAVSDK can just listen to any UDP arriving at that default port.
 
-```sh
-ping 169.254.21.183
-
-PING 169.254.21.183 56 bytes of data
-56 bytes from 169.254.21.183: icmp_seq=0 time=0 ms
-56 bytes from 169.254.21.183: icmp_seq=1 time=0 ms
-56 bytes from 169.254.21.183: icmp_seq=2 time=0 ms
-56 bytes from 169.254.21.183: icmp_seq=3 time=0 ms
-56 bytes from 169.254.21.183: icmp_seq=4 time=0 ms
-56 bytes from 169.254.21.183: icmp_seq=5 time=0 ms
-56 bytes from 169.254.21.183: icmp_seq=6 time=0 ms
-56 bytes from 169.254.21.183: icmp_seq=7 time=0 ms
-56 bytes from 169.254.21.183: icmp_seq=8 time=0 ms
-56 bytes from 169.254.21.183: icmp_seq=9 time=0 ms
-10 packets transmitted, 10 received, 0% packet loss, time 10010 ms
-```
+To run a MAVSDK example, install mavsdk via pip, and try out an example from [MAVSDK-Python/examples](https://github.com/mavlink/MAVSDK-Python/tree/main/examples).
 
-#### MAVLink/MAVSDK Test
+#### XRCE-Client Ethernet Setup
 
-For this, we need to set the MAVLink instance to send traffic to the CM4's IP address:
+Now it is needed to enable MAVLink on the PX4 `Ethernet` port and enable `XRCE-DDS`.
```

Comment:
What are you doing here? I THINK you are setting up MAVLink to run on the wired connection to TELEM2, and XRCE-DDS to connect to the Ethernet link you just set up right?

Reply:
> What are you doing here? I THINK you are setting up MAVLink to run on the wired connection to TELEM2, and XRCE-DDS to connect to the Ethernet link you just set up right?

I am enabling Ethernet on MAV_2 here.

Reply:
So you're setting both mavlink and XRCE-DDS on the same local ethernet serial port?:

```
param set MAV_2_CONFIG     1000        # Configure Ethernet port
param set UXRCE_DDS_CFG    1000   
```

I don't think that is allowed - it shouldn't work as only one thing should own a serial port at a time.

Don't we normally have mavlink over a separate dedicated internal serial link?

Reply:
You are absolutely right Hamish! My bad in tests.